api/mfx_dispatch/linux/device_ids.h

```diff
@@ -0,0 +1,471 @@
+    { 0x5A4C, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A50, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A54, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A59, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A5C, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A41, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A44, MFX_HW_CNL, GT_INTEGRATED },
+    { 0x5A49, MFX_HW_CNL, GT_INTEGRATED },
+
+    /* ICL LP */
+    { 0xFF05, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A50, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A51, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A52, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A53, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A54, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A56, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A57, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A58, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A59, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A5A, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A5B, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A5C, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A5D, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A70, MFX_HW_ICL_LP, GT_INTEGRATED },
+    { 0x8A71, MFX_HW_ICL_LP, GT_INTEGRATED },  // GT05, but 1 ok in this context
+
+    /* JSL */
+    { 0x4E51, MFX_HW_JSL, GT_INTEGRATED },
+    { 0x4E55, MFX_HW_JSL, GT_INTEGRATED },
+    { 0x4E61, MFX_HW_JSL, GT_INTEGRATED },
+    { 0x4E71, MFX_HW_JSL, GT_INTEGRATED },
+
+    /* EHL */
+    { 0x4500, MFX_HW_EHL, GT_INTEGRATED },
+    { 0x4541, MFX_HW_EHL, GT_INTEGRATED },
+    { 0x4551, MFX_HW_EHL, GT_INTEGRATED },
+    { 0x4555, MFX_HW_EHL, GT_INTEGRATED },
+    { 0x4569, MFX_HW_EHL, GT_INTEGRATED },
+    { 0x4571, MFX_HW_EHL, GT_INTEGRATED },
+
+    /* TGL */
+    { 0x9A40, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A49, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A59, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A60, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A68, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A70, MFX_HW_TGL_LP, GT_INTEGRATED },
+    { 0x9A78, MFX_HW_TGL_LP, GT_INTEGRATED },
+
+    /* DG1/SG1 */
+    { 0x4905, MFX_HW_DG1, GT_DISCRETE },
+    { 0x4906, MFX_HW_DG1, GT_DISCRETE },
+    { 0x4907, MFX_HW_DG1, GT_DISCRETE },
+    { 0x4908, MFX_HW_DG1, GT_DISCRETE },
+
+    /* RKL */
+    { 0x4C80, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+    { 0x4C8A, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+    { 0x4C81, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+    { 0x4C8B, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+    { 0x4C90, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+    { 0x4C9A, MFX_HW_RKL, GT_INTEGRATED }, // RKL-S
+
+    /* ADL-S */
+    { 0x4600, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4680, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4681, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4683, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4690, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4691, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4693, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4698, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4699, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
```

Comment:
Why ADL defined?  I thought VPL was going to be loaded on ADL.  But only unknown loads VPL.

Reply:
There are 2 platforms for ADL: ADL-S and ADL-P. Mediasdk runtime ends up with ADL-S (inclusively), but it does not support ADL-P. FYI: VPL runtime supports platforms starting from TGL (aka gen 12 family). The story here is somewhat complicated, we plan to provide some documentation around that. Hold on.

Reply:
Does ADL-S and ADL-P have the same GPU Gen?

Reply:
Yes, It's Gen12

Reply:
Sorry, what I meant was, is it same minor rev, too (12.2)?

Reply:
Yes

Reply:
@uartie , please see reply from @daleksan  below.